### PR TITLE
Adds Comment Attachments

### PR DESCRIPTION
**Pull request by lorenjohnson**
* Goes with https://github.com/Hylozoic/hylo-node/pull/519
* Adds multiple file attachments to Post Editor as well (https://github.com/Hylozoic/hylo-evo/issues/555)

### CODE REVIEW

File: src/store/actions/createComment.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { CREATE_COMMENT } from 'store/constants'
```
Comment (tibetsprague) [inline src/store/actions/createComment.js:1]: lets chat about the strategy for actions moving forward. im not totally convinced yet about moving everything into actions... probably better but worth a chat

File: src/store/actions/createComment.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      variables: {
+        postId,
+        text,
+        attachments
+      }
+    },
+    meta: {
+      optimistic: true,
+      extractModel: 'Comment',
+      tempId: uniqueId(`post${postId}_`),
```
Comment (tibetsprague) [inline src/store/actions/createComment.js:24]: what is this for?

File: src/components/AttachmentManager/AttachmentManager.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    ) {
+      this.props.loadAttachments()
+    }
+  }
+
+  componentWillUnmount () {
+    this.props.clearAttachments()
+  }
+
+  clearAttachments = () => this.props.clearAttachments()
```
[the remark this answers is not in the document]
Comment (lorenjohnson) [reply src/components/AttachmentManager/AttachmentManager.js:67]: Ah yeah, good catch. I had it there when I was playing with a different implementation. Will remove.

File: src/components/AttachmentManager/AttachmentManager.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+  </div>
+}
+
+export function FilePreview ({ attachment, removeFile, fileSize }) {
+  const filename = path.basename(attachment.url)
+  return <div styleName='file-preview'>
+    <Icon name='Document' styleName='icon-document' />
+    <div styleName='file-name'>{decodeURIComponent(filename)}</div>
+    {fileSize && <div styleName='file-size'>{fileSize}</div>}
+    {removeFile && <Icon name='Ex' styleName='remove-file' onClick={removeFile} />}
```
[the remark this answers is not in the document]
Comment (lorenjohnson) [reply src/components/AttachmentManager/AttachmentManager.js:192]: Yep, same here. Vestige of an abandoned implementation, will remove.

File: src/components/AttachmentManager/AttachmentManager.store.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  attachmentType
+}) {
+  const allAttachments = getOr([], [MODULE_NAME, makeAttachmentKey(type, id)], state)
+
+  if (attachmentType) return filter({ attachmentType }, allAttachments)
+
+  return allAttachments
+}
+
+export const getAttachmentsFromObject = ormCreateSelector(
```
[the remark this answers is not in the document]
Comment (lorenjohnson) [reply src/components/AttachmentManager/AttachmentManager.store.js:78]: `...FromObject` meaning from it gets the Attachments for any ReduxORM object matching the type and id pair.  Will comment the same.

File: src/components/PostEditor/PostEditor.connector.js [hunk 6]
```diff
@@ -71,39 +73,43 @@ export function mapStateToProps (state, props) {
...
-    removeLinkPreview: () => dispatch(removeLinkPreview()),
-    clearLinkPreview: () => dispatch(clearLinkPreview()),
-    updatePost: postParams => dispatch(updatePost(postParams)),
-    createPost: postParams => dispatch(createPost(postParams)),
-    createProject: postParams => dispatch(createProject(postParams)),
     goToUrl: url => dispatch(push(url)),
-    addImage: url => dispatch(addAttachment(url, 'image')),
-    addFile: url => dispatch(addAttachment(url, 'file')),
-    setAnnouncement: bool => dispatch(setAnnouncement(bool))
+    ...bindActionCreators({
```
[the remark this answers is not in the document]
Comment (lorenjohnson) [reply src/components/PostEditor/PostEditor.connector.js:104]: The `mapDispatchToProps` param to `connect` can either be a function which returns `dispatch` wrapped functions or a simple object which will get `bindActionCreators(...)` applied to it automatically. When needing to have a mix of both you need to do this or just wrap dispatch on even the simple cases. I like to pull in `bindActionCreators({...})` to make explicit all the actions which aren't getting modified on the way into the component such that if any special cases are eliminated you can go back to the simple object notation... Hope that makes sense.

File: src/components/PostEditor/PostEditor.connector.test.js [hunk 7]
```diff
@@ -36,10 +36,9 @@ beforeAll(() => {
     pending: {
       FETCH_LINK_PREVIEW: false
     },
-    AttachmentManager: {
-      image: [],
-      file: []
-    }
+    // AttachmentManager: {
```
[the remark this answers is not in the document]
Comment (lorenjohnson) [reply src/components/PostEditor/PostEditor.connector.test.js:39]: Fo sho.

File: src/graphql/fragments/postFieldsFragment.js [hunk 8]
```diff
@@ -32,8 +32,10 @@ const postFieldsFragment = withComments => `
         avatarUrl
       }
       attachments {
-        id
+        type
         url
+        position
+        id
```
[the remark this answers is not in the document]
Comment (lorenjohnson) [reply src/graphql/fragments/postFieldsFragment.js:38]: I'm so good with anything consistent, and happy to alpha it all going forward. `id` got moved down the list in this case as it's the `id` of the corresponding `Media` record in `hylo-node` which as far as I can tell is never used in the display or updating of attachments in the front-end. Probably could just be removed from the query but will just re-sort the keys for now.

File: src/store/actions/createComment.js (hunk 1 above)
Comment (lorenjohnson) [reply src/store/actions/createComment.js:1]: For sure. This is a gray area, but I also fairly strictly move redux things out of a local store file if they are operating on the store outside of the context of that local store AND they are used outside of that component at least once. The more general thing going on here though is I'm trying to surface all the redux and graphql in one place to simplify later transitions away from the particular use of either. Happy to discuss. Needless to say I've a lot of things in motion in me from being in this code for 3 years and some of them you may find value in, some of them you may wish to go a different way with...

File: src/store/actions/createComment.js (hunk 2 above)
Comment (lorenjohnson) [reply src/store/actions/createComment.js:24]: That predated my work and could speculate, but essentially I just left it as is.

**Pull request merged**